src/ngraph/node.cpp

```diff
@@ -334,9 +334,54 @@ const std::set<std::shared_ptr<Node>>& Node::get_control_dependencies() const
     return m_control_dependencies;
 }
 
+const std::set<Node*>& Node::get_control_dependents() const
+{
+    return m_control_dependents;
+}
+
 void Node::add_control_dependency(std::shared_ptr<Node> node)
 {
     m_control_dependencies.insert(node);
+    node->m_control_dependents.insert(this);
+}
+
+void Node::add_node_control_dependencies(std::shared_ptr<Node> source_node)
```

Comment:
I think it'd be better if this and `add_node_control_dependents` were free functions.